example/lib/pages/sign-up/sign-up-page-state.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:example/pages/routes.dart';
+import 'package:example/pages/sign-up/sign-up-page.dart';
+import 'package:example/sdk/services/user-service/user-service.dart';
+import 'package:floater/floater.dart';
+import 'package:flutter/material.dart';
+
+class SignUpPageState extends WidgetStateBase<SignUpPage> {
+  final _navigator = NavigationService.instance.retrieveNavigator("/");
+  final _userService = ServiceLocator.instance.resolve<UserService>();
+
+  late Validator<SignUpPageState> _validator;
+
+  String _firstName = "";
+  String _lastName = "";
+  String _phone = "";
+  String _password = "";
+
+  String get firstName => this._firstName;
+  set firstName(String value) => (this.._firstName = value).triggerStateChange();
+
+  String? get lastName => this._lastName;
+  set lastName(String? value) => (this.._lastName = value ?? "").triggerStateChange();
+
+  String get phone => this._phone;
+  set phone(String value) => (this.._phone = value).triggerStateChange();
+
+  String get password => this._password;
+  set password(String value) => (this.._password = value).triggerStateChange();
+
+  bool get hasErrors => this._validator.hasErrors;
+  ValidationErrors get errors => this._validator.errors;
+
+  SignUpPageState() : super() {
+    this._createValidator();
+    this.onStateChange(() {
+      this._validate();
+    });
+  }
+
+  Future<void> signUp() async {
+    this._validator.enable();
+    if (!this._validate()) {
+      this.triggerStateChange();
+      return;
+    }
+    this.showLoading();
+    try {
+      await this._userService.createUser(this.firstName, this.lastName, this.phone, this.password);
+      this._navigator.popUntil((route) => route.isFirst);
```

Comment:
?

Reply:
since i am supporting going back, i don't want the user to go back to sign up from login, so i want to clear the stack and go to login as new, it was really weird,

Reply:
you should pop all the pages don't keep the first one

Reply:
you should pop all the pages